### PR TITLE
Action v11 with fixes and modern setup

### PR DESCRIPTION
**Pull request by d-koppenhagen**
@cmgriffing would you mind to check this out? I modernized the whole setup of the repo based on:
https://github.com/actions/typescript-action

I fixed also some issues and updated the documentation.

### CODE REVIEW

**Comment by d-koppenhagen** [comment]
I think some pipeline steps are failing since I have no valid access_token. @cmgriffing probably this works I you are running the pipeline.

**Comment by cmgriffing** [comment]
Nice, thanks. I think I made a mistake in approving and merging some other commits first. Could you fix those conflicts?
After that, I can just merge it in, and we can see if the tests pass then.

File: .github/linters/.eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  {
+    "camelcase": "off",
+    "eslint-comments/no-use": "off",
+    "eslint-comments/no-unused-disable": "off",
+    "i18n-text/no-en": "off",
+    "import/no-namespace": "off",
+    "no-console": "off",
+    "no-unused-vars": "off",
+    "prettier/prettier": "error",
+    "semi": "off",
```
Comment (cmgriffing) [inline .github/linters/.eslintrc.yml:47]: I think, I would prefer to keep the semicolons for now.

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "printWidth": 80,
+  "tabWidth": 2,
+  "useTabs": false,
+  "semi": false,
```
Comment (cmgriffing) [inline .prettierrc.json:5]: Also here, I would prefer the semicolons be left on.

**Comment by cmgriffing** [comment]
Merged, let's double check things work.

THANK YOU SO MUCH @d-koppenhagen !!!

**Comment by d-koppenhagen** [comment]
awesome. Once you checked, can you draft the v11 release?

**Comment by d-koppenhagen** [comment]
I tested it with my blog using the SHA of the Commit after the merge and everything seems to work:

**Pipeline:**
https://github.com/d-koppenhagen/k9n.dev/actions/runs/7015168587/job/19084033853

**Related Commit:**
https://github.com/d-koppenhagen/k9n.dev/commit/0afa75cb975b596bd40ebb696e8468be2da7eb56

**Pull request merged**